src/main/java/uk/gov/hmcts/reform/ccd/documentam/client/dmstore/DocumentStoreClient.java

```diff
@@ -0,0 +1,181 @@
+package uk.gov.hmcts.reform.ccd.documentam.client.dmstore;
+
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.ccd.documentam.ApplicationParams;
+import uk.gov.hmcts.reform.ccd.documentam.apihelper.Constants;
+import uk.gov.hmcts.reform.ccd.documentam.dto.DocumentUploadRequest;
+import uk.gov.hmcts.reform.ccd.documentam.exception.ResourceNotFoundException;
+import uk.gov.hmcts.reform.ccd.documentam.exception.ServiceException;
+import uk.gov.hmcts.reform.ccd.documentam.model.DmTtlRequest;
+import uk.gov.hmcts.reform.ccd.documentam.model.DmUploadResponse;
+import uk.gov.hmcts.reform.ccd.documentam.model.Document;
+import uk.gov.hmcts.reform.ccd.documentam.model.PatchDocumentResponse;
+import uk.gov.hmcts.reform.ccd.documentam.model.UpdateDocumentsCommand;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.springframework.http.HttpMethod.GET;
+import static uk.gov.hmcts.reform.ccd.documentam.apihelper.Constants.DM_DATE_TIME_FORMATTER;
+import static uk.gov.hmcts.reform.ccd.documentam.apihelper.Constants.EXCEPTION_ERROR_ON_DOCUMENT_MESSAGE;
+import static uk.gov.hmcts.reform.ccd.documentam.apihelper.Constants.RESOURCE_NOT_FOUND;
+
+@Named
+public class DocumentStoreClient {
+
+    private final RestTemplate restTemplate;
+    private final ApplicationParams applicationParams;
+
+    @Inject
+    public DocumentStoreClient(final RestTemplate restTemplate,
+                               final ApplicationParams applicationParams) {
+        this.restTemplate = restTemplate;
+        this.applicationParams = applicationParams;
+    }
+
+    public Optional<Document> getDocument(final UUID documentId) {
+        try {
+            final Document document = restTemplate.getForObject(
+                String.format("%s/documents/%s", applicationParams.getDocumentURL(), documentId),
+                Document.class
+            );
+
+            return Optional.ofNullable(document);
+        } catch (HttpClientErrorException exception) {
+            if (HttpStatus.NOT_FOUND.equals(exception.getStatusCode())) {
+                return Optional.empty();
+            }
+            throw new ServiceException(
```

Comment:
Could these exceptions not be handled in the exception handler `CaseDocumentControllerAdvice.java` ?

Reply:
Yes, that would make sense in the instances where we don't do anything but re-wrap the exception and re-throw.  Not sure if it would be done though, as the plan seems to be to a step away from rest-template altogether